### DISCUSSION TITLE
How to display Status & Progressbar Together?

### DISCUSSION THREAD

**Discussion by twhughes** [Textualize/rich#1594] [Q&A]
I'm trying to come up with a good solution for displaying a progressbar for each status in a list.

```python
for status in statuses:
    # display the current status with spinner
    # display a progressbar for the current status progress.
```

For a concrete example.

```python
import time
from rich.progress import Progress
from rich.console import Console

# outer status bar
console = Console()
for status_name in ('status1', 'status2', 'status3'):
    with console.status(f"[bold green]Status = {status_name}") as status:

        # progressbar for status
        with Progress() as progress:
            task = progress.add_task('running', total=100)

            for i in range(100):
                progress.update(task, advance=1)
                time.sleep(0.05)
```

However, this solution seems to not render very cleanly as when the progressbar updates, the status plot will disappear and reappear.  Is there a better solution for this kind of problem?  Perhaps I need to split the status and progressbar as a table view?

Thanks!

**Comment by jfthuong** [comment]
Here is some code that could help:

The idea is:
1. Create both `status` and `Progress` objects first
2. Include them in a group (previously called `RenderableGroup` and now just `Group`), itself inside a `Live` object
3. Optionally, we could insert the group inside a `Panel` to add a frame
4. At each loop, we can update if needed / reset / add tasks as needed...

... Updated on 2025-03-16 thanks to 2 comments from @mcourteaux 

```python
import time
from rich.progress import Progress
from rich.console import Console, Group
from rich.live import Live
from rich.panel import Panel

console = Console()

# outer status bar and progress bar
status = console.status("Not started")
progress = Progress(transient=True)

with Live(Panel(Group(status, progress)), console=console):  # Optionally: you can remove the `Panel(...)` so just `Live(Group(...))`
    for status_name in ('status1', 'status2', 'status3'):
        status.update(f"[bold green]Status = {status_name}")
        task_id = progress.add_task(f"Working on task for {status_name}")

        for i in progress.track(range(25), task_id=task_id):
            time.sleep(0.1)
        progress.remove_task(task_id)

    status.update("[bold green]Status = Done")
```

This looks like this during the run
![image](https://user-images.githubusercontent.com/1401809/174727294-ca781944-feb3-4dbb-a7c9-fffb766dd35a.png)

And like that at the end:
![image](https://user-images.githubusercontent.com/1401809/174727256-d4050303-cdca-4758-a9d5-6d6303bd20e8.png)


... and if you want the status bar below (which would look nicer), just do:

```python
...
with Live(Panel(Group(progress, status))):
    ...
```

![image](https://user-images.githubusercontent.com/1401809/174727526-a3bfd82a-6bba-409a-978e-4da159540e0c.png)

**Reply by mcourteaux** [reply]
Printing to the (parent) `rich.console.Console` while this `Live` is active, overwrites the printed line. I managed to resolve this by passing the `console` to the `Live` constructor as well.

**Reply by mcourteaux** [reply]
Note that the `Panel` is optional.

**Reply by jfthuong** [reply]
> Note that the `Panel` is optional.

Indeed, `Panel` is optional, but it makes it nicer (especially since the elements are grouped together)

... without `Panel`:
![image](https://github.com/user-attachments/assets/40a4234e-24ea-4826-b034-dc8f7bd3eacb)

... with `Panel`:
![image](https://github.com/user-attachments/assets/51b12b49-54fc-4901-8490-43a7cb0fb2ea)

Thanks for the addition (especially the console): I have updated the code above
